@here/harp-mapview-decoder/test/DataProviderTest.ts

```diff
@@ -60,4 +71,44 @@ describe("DataProvider", function() {
         dataProvider.unregister(clients[1]);
         expect(disposeSpy.calledOnce).is.true;
     });
+
+    it("calls connect only on first registered client", async function() {
+        const dataProvider = new TimedFakeDataProvider();
+        const connectSpy = sinon.spy(dataProvider, "connect");
+
+        await dataProvider.register(clients[0]);
+        expect(connectSpy.calledOnce).is.true;
+
+        await dataProvider.register(clients[0]);
+        await dataProvider.register(clients[1]);
+        expect(connectSpy.calledOnce).is.true;
+    });
+
+    it("calls connect only on first registered client", async function() {
```

Comment:
this test is duplicating the first part of the one above, also has the same name...

Reply:
Yes, it covers the same functionality as the first one, removed.